trio/_file_io.py

```diff
@@ -48,7 +74,132 @@
 }
 
 
-class AsyncIOWrapper(AsyncResource):
+FileT = TypeVar("FileT")
+FileT_co = TypeVar("FileT_co", covariant=True)
+T = TypeVar("T")
+T_co = TypeVar("T_co", covariant=True)
+T_contra = TypeVar("T_contra", contravariant=True)
+AnyStr_co = TypeVar("AnyStr_co", str, bytes, covariant=True)
+AnyStr_contra = TypeVar("AnyStr_contra", str, bytes, contravariant=True)
```

Comment:
These types might also warrant some comments. One day I'll get comfortable with co- & contravariant - but some brief comments here saying which ones are used where and why will probably help readers and any future people trying to modify it.

Reply:
I mainly picked/needed them because protocols complain if you don't have the right ones :)